mmv1/templates/terraform/pre_update/notebooks_runtime.go.erb

```diff
@@ -0,0 +1,60 @@
+// remove virtualMachine from updateMask
+callSwitch := false
+for i, field := range updateMask {
+    if field == "virtualMachine" {
+      callSwitch = true
+        updateMask = append(updateMask[:i], updateMask[i+1:]...)
+        break
+    }
+}
+
+// reconstruct url
+url, err = tpgresource.ReplaceVars(d, config, "{{NotebooksBasePath}}projects/{{project}}/locations/{{location}}/runtimes/{{name}}")
+if err != nil {
+    return err
+}
+url, err = transport_tpg.AddQueryParams(url, map[string]string{"updateMask": strings.Join(updateMask, ",")})
+if err != nil {
+    return err
+}
+
+if callSwitch {
+  state := d.Get("state").(string)
+  if state == "INITIALIZING" {
+    time.Sleep(300 * time.Second)
+	}
+  switchURL, err := tpgresource.ReplaceVars(d, config, "{{NotebooksBasePath}}projects/{{project}}/locations/{{location}}/runtimes/{{name}}:switch")
+	  if err != nil {
+		  return err
+	  }
+  log.Printf("[DEBUG] Switching Runtime: %q", d.Id())
+
+  switchObj := make(map[string]interface{})
+  switchObj["machineType"] = d.Get("virtual_machine.0.virtual_machine_config.0.machine_type")
+  acceleratorConfig := make(map[string]interface{})
+  acceleratorConfig["coreCount"] = d.Get("virtual_machine.0.virtual_machine_config.0.accelerator_config.0.core_count")
+  acceleratorConfig["type"] = d.Get("virtual_machine.0.virtual_machine_config.0.accelerator_config.0.type")
```

Comment:
What happens if some of these values aren't set ? I think we should  be selectively including them by ensuring they are non-empty

Reply:
Looks like  acceleratorConfig["type"] needs to be of an enum type. Any references on how this is done?

Error: Error switching Runtime: googleapi: Error 400: Invalid value at 'accelerator_config.type' (type.googleapis.com/google.cloud.notebooks.v1.RuntimeAcceleratorConfig.AcceleratorType), ""
        Details:
        [
          {
            "@type": "type.googleapis.com/google.rpc.BadRequest",
            "fieldViolations": [
              {
                "description": "Invalid value at 'accelerator_config.type' (type.googleapis.com/google.cloud.notebooks.v1.RuntimeAcceleratorConfig.AcceleratorType), \"\"",
                "field": "accelerator_config.type"
              }
            ]
          }
        ]

Reply:
https://github.com/GoogleCloudPlatform/magic-modules/blob/a5036e705278daac790cab6f7127038cf924155a/mmv1/products/appengine/FirewallRule.yaml#L64

there is an Enum type we can use. Although we should be careful because it has more maintenance overhead in utilizing it (if a new value is supported at the api level we need a manual update here). Usually we just use strings, let the API fail and link to documented values in the description.

Reply:
Noted. Fixed the issue caused by empty configs.